### DISCUSSION TITLE
magit-insert-section `processbuf`?

### DISCUSSION THREAD

**Discussion by Silex** [magit/magit#5190] [Support] [answered]
Hello,

It's probably not a bug and just me that is stupid about what's going on, but I cannot understand where `processbuf` comes from at https://github.com/magit/magit/blob/main/lisp/magit-process.el#L347

Is it just a "string" used by the magit-insert-section macro to figure what kind of section it should create? Or maybe some undocumented emacs type?

**Comment by tarsius** [comment, marked as answer]
Increasingly the *type* corresponds to a class (defined using `defclass`), but for historic and pragmatic reasons that is not always the case.  If there is not specific class, then the `magit-section` class is used. `magit-insert-section` selects the local keymap based on the type and it also stores the type in an overlay. Commands can use `magit-section-match` to determine their behavior based on the section at point (most often its type and value).

See https://magit.vc/manual/magit-section/Creating-Sections.html and https://magit.vc/manual/magit-section/Matching-Functions.html, and grep magit for `magit-section-match` and `magit-section-cond` for examples.

**Reply by Silex** [reply]
Thanks! Will dig into it more. Also I note that Github now has discussions, neat.
